batch/batch/worker/worker.py

```diff
@@ -2234,6 +2230,11 @@ async def cleanup(self):
                     await self.jvm.cloudfuse_mount_manager.unmount(mount_path, user=self.user, bucket=bucket)
                     config['mounted'] = False
 
+        with open('/proc/mounts', 'r') as f:
+            output = f.read()
+            if self.cloudfuse_base_path() in output:
```

Comment:
I think my question is won't this fuse path remain alive if there's other jobs reading the data? That's why I had the check after we actually call `fusermount`.

Reply:
This is not a fuse path, it's a bind mount path which is the thing that the JVMJob actually owns. Here's an example:

1. JVM 1 wants bucket `foo` at `/cloudfuse/JVM-1/foo`
2. FUSE cache manager makes two mounts:
  - FUSE mount at `/cloudfuse/read_only/abc123/foo`
  - Bind mounts `/cloudfuse/read_only/abc123/foo` to `/cloudfuse/JVM-1/foo`
3. JVM 2 wants bucket `foo` at `/cloudfuse/JVM-2/foo`
4. FUSE cache manager reuses the underlying FUSE mount and bind mounts `/cloudfuse/read_only/abc123/foo` to `/cloudfuse/JVM-2/foo`.
5. JVM 1 finishes and wants to clean up, it calls `cloudfuse_mount_manager.unmount`
6. FUSE cache manager unmounts the bind mount at `/cloudfuse/JVM-1/foo`. The underyling FUSE mount is still being used by another job, so it doesn't do anything else

At this point, the only thing JVM 1 needs to validate to make sure that it is properly cleaned up is that the bind mount that it owns (whatever is under `/cloudfuse/JVM-1`) is gone. Without that bind mount it has no connection to the underlying FUSE mount created by the cache manager.

Reply:
Thanks! This was really helpful.